corehq/motech/repeaters/models.py

```diff
@@ -614,8 +614,10 @@ def allowed_to_forward(self, payload):
 
         # Exclude extension cases where the host is also a case type that this repeater
         # would act on since they get forwarded along with their host
-        host_index = payload.get_index(CASE_INDEX_IDENTIFIER_HOST)
-        return not host_index or host_index.referenced_type not in self.white_listed_case_types
+        host_indices = payload.get_indices(relationship=CASE_INDEX_EXTENSION)
+        for host_index in host_indices:
+            if host_index.referenced_type in self.white_listed_case_types:
+                return False
```

Comment:
I think you're missing a
```
        return True
```
on the next line.

Reply:
Yea, this is actually broken anyway and fixed here: https://github.com/dimagi/commcare-hq/pull/31616